src/middlewares/requiredFieldsMiddleware.ts

```diff
@@ -38,7 +38,22 @@ const requiredFields = ({ getState, dispatch }: MiddlewareAPI<Dispatch<AnyAction
             state.view.widgets
             .filter(item => item.bcName === widget.bcName)
             .forEach(item => {
-                item.fields.forEach((widgetField: WidgetField) => {
+                const itemFieldsCalc: object[] = item.fields
+                if (item.fields) {
+                    item.fields.forEach((block: object | WidgetFieldBlock<object>) => {
+                        /**
+                         * block check
+                         * @param itemObject
+                         */
+                        function isWidgetFieldBlock(itemObject: any): itemObject is WidgetFieldBlock<object> {
```

Comment:
This function is importable from [widget.ts](https://github.com/tesler-platform/tesler-ui/blob/ce0a99cbb6a249e7265bdb254d49d9d0e3aca858/src/interfaces/widget.ts#L241):

```tsx
import {WidgetField, WidgetFieldBlock, isWidgetFieldBlock} from '../interfaces/widget'
```